distribution/xfer/download.go

```diff
@@ -62,6 +63,9 @@ type DownloadDescriptor interface {
 	// if it is unknown (for example, if it has not been downloaded
 	// before).
 	DiffID() (layer.DiffID, error)
+	// ForeignSource returns nil or the foreign layer's information and
+	// source.
+	ForeignSource() *distribution.Descriptor
```

Comment:
Why not just expose a descriptor for the layer and then have method to indicate whether or not it is foreign?